Userland/Libraries/LibJS/Runtime/RegExpConstructor.cpp

```diff
@@ -8,9 +8,20 @@
 #include <LibJS/Runtime/GlobalObject.h>
 #include <LibJS/Runtime/RegExpConstructor.h>
 #include <LibJS/Runtime/RegExpObject.h>
+#include <LibJS/Runtime/Value.h>
 
 namespace JS {
 
+void RegExpStaticProperties::invalid()
+{
+    m_input = Utf16String(String::empty().view());
+    m_last_match = Utf16String(String::empty().view());
+    m_last_paren = Utf16String(String::empty().view());
+    m_left_context = Utf16String(String::empty().view());
+    m_right_context = Utf16String(String::empty().view());
```

Comment:
We usually use UTF-8 strings across the board in LibJS and only convert them on the fly when needed. Any reason to not do the same here?

Reply:
They come from `regexp_builtin_exec` and the last argument type is `Utf16String`

```cpp
// 22.2.5.2.2 RegExpBuiltinExec ( R, S ), https://tc39.es/ecma262/#sec-regexpbuiltinexec
static ThrowCompletionOr<Value> regexp_builtin_exec(VM& vm, RegExpObject& regexp_object, Utf16String string)
{
```

Reply:
I see, but I'm not convinced we need to propagate that. What's wrong with changing this:

```cpp
// 12. Set the value of C’s [[RegExpRightContext]] internal slot to a String whose length is len - endIndex and containing the code units from S with indices endIndex through len - 1, in ascending order.
auto const& right_context = Utf16String(string.view().substring_view(end_index, len - end_index));
m_legacy_static_properties.set_right_context(right_context);
```

to this:

```cpp
// 12. Set the value of C’s [[RegExpRightContext]] internal slot to a String whose length is len - endIndex and containing the code units from S with indices endIndex through len - 1, in ascending order.
auto right_context = string.view().substring_view(end_index, len - end_index).to_utf8();
m_legacy_static_properties.set_right_context(move(right_context));
// Declaration: set_right_context(String);
```

Reply:
ok